src/BatteryStateOfChargeService.cpp

```diff
@@ -16,21 +24,68 @@ BatteryStateOfChargeService::~BatteryStateOfChargeService()
 
 double BatteryStateOfChargeService::totalAmpHoursUsed() const
 {
-    return 0.0;
+    return BATTERY_AMP_HOUR_CAPACITY-ampHours ;
 }
 
 bool BatteryStateOfChargeService::isCharging() const
 {
+    if(previousCurrent<0)
+    {
     return false;
+    }
+    else
+    {
+       return true;
+    }
 }
 
 QTime BatteryStateOfChargeService::timeWhenChargedOrDepleted() const
 {
-    return QTime::currentTime();
+    double hoursLeftOnCharge;
+    double minutesLeftOnCharge;
+    double secondsLeftOnCharge;
+
+    double timeRemaining = (BATTERY_AMP_HOUR_CAPACITY-ampHours)/previousCurrent ;
+
+    timeRemaining = qAbs(timeRemaining);
+
+    hoursLeftOnCharge = timeRemaining;
+    int hours = int(timeRemaining);
+    int h = hours;
+
+    minutesLeftOnCharge = (hoursLeftOnCharge-h)*60;
+    int m = (int)minutesLeftOnCharge;
+
+    secondsLeftOnCharge = (minutesLeftOnCharge - m)*60;
+    int s = (int)secondsLeftOnCharge;
+
+    int z = (secondsLeftOnCharge-s)*1000;
```

Comment:
I'd suggest this whole section of constructing a `QTime` can be done using the built-in function `QTime.addMSecsTo`.